Source/Windows/ARM64EC/Module.cpp

```diff
@@ -196,6 +194,31 @@ void PatchCallChecker() {
   WriteModuleRVA(Module, CHPEMetadata->__os_arm64x_dispatch_icall_cfg, &CheckCall);
 }
 
+// Fills in the syscall numbers necessary to call *Native variants of syscalls from FEX under wine.
+void ParseWineSyscallNumbers(HMODULE NtDll) {
+  ULONG Size;
+  const auto* Exports = reinterpret_cast<IMAGE_EXPORT_DIRECTORY*>(RtlImageDirectoryEntryToData(NtDll, true, IMAGE_DIRECTORY_ENTRY_EXPORT, &Size));
+  const auto* NameTable = reinterpret_cast<uint32_t*>(NtDllBase + Exports->AddressOfNames);
+
+  // Wine orders syscalls alphabetically, take advantage of that to find the syscall indices for those which we need to
+  // manually issue. Note that all functions starting with Nt besides NtGetTickCount are syscalls.
+  uint32_t CurSyscallId = 0;
+  for (uint32_t Idx = 0; Idx < Exports->NumberOfNames; Idx++) {
+    const char* Name = reinterpret_cast<const char*>(NtDllBase + NameTable[Idx]);
+    if (Name[0] == 'N' && Name[1] == 't' && strcmp(Name, "NtGetTickCount")) {
+      if (!strcmp(Name, "NtContinue")) {
+        WineNtContinueSyscallId = CurSyscallId;
+      } else if (!strcmp(Name, "NtAllocateVirtualMemory")) {
+        WineNtAllocateVirtualMemorySyscallId = CurSyscallId;
+      } else if (!strcmp(Name, "NtProtectVirtualMemory")) {
+        WineNtProtectVirtualMemorySyscallId = CurSyscallId;
+      }
+
+      CurSyscallId++;
+    }
+  }
```

Comment:
Checking for string inequality by casting strcmp to bool is unnecessarily clever and seems to have already been misinterpreted once.

Using string_view avoids this issue:
```suggestion
    std::string_view Name = reinterpret_cast<const char*>(NtDllBase + NameTable[Idx]);
    if (Name.starts_with("Nt") && Name != "NtGetTickCount") {
      if (Name == "NtContinue") {
        WineNtContinueSyscallId = CurSyscallId;
      }
      // ...

      CurSyscallId++;
    }
  }
```

Reply:
right, was avoiding string view out of caution here so it was very clear exactly what ntdll library calls were made (as is said in comments, until our call checker is initialised we have to be incredibly cautious of what we call as who-knows-what exports could be patched), there's a small chance that some called by InitSyscalls will need to be reimplemented purely within FEX as more drm schemes are tested. That being said checking the resulting binary it seems fine in terms of export usage, so will let you judge if the readability benefit outweighs the above (will also  namespace such functions together so it's harder to miss the context on their limitations)

Reply:
*not* using `string_view` is also fine, but if we need to compromise for weaker abstractions due to this extra level of caution, we should put an equal effort into compensating the cost of this.

Using an early continue and an explicit comparison, along with a comment to express intent, would achieve this:
```
// Skip non-Nt syscalls and NtGetTickCount
if (Name[0] != 'N' || Name[1] != 't' || strcmp(Name, "NtGetTickCount") == 0){
  continue;
}
```

Reply:
made the != 0 explicit and added more info to comments